core/src/main/scala/org/bitcoins/core/protocol/ln/routing/Route.scala

```diff
@@ -0,0 +1,13 @@
+package org.bitcoins.core.protocol.ln.routing
+
+import org.bitcoins.core.protocol.ln.channel.ShortChannelId
+import org.bitcoins.core.protocol.ln.node.NodeId
+
+/**
+  * Represent differet types of LN routes. Supports node and channel routes.
+  */
+sealed trait Route
```

Comment:
Maybe consider this? I think we can get rid of some of the `.asInstanceOf[]` then 

```scala
sealed trait Route[T] {
  def ids: Vector[T]
}

case class NodeRoute(ids: Vector[NodeId]) extends Route[NodeId]

case class ChannelRoute(ids: Vector[ShortChannelId])
    extends Route[ShortChannelId]
```